### PR TITLE
ci: Extend python base version for test cases

### PR DESCRIPTION
**Pull request by sudohainguyen**
**What this PR does / why we need it**:
Following up with [NEP](https://numpy.org/neps/nep-0029-deprecation_policy.html) and many libraries are dropping py38 support
We need to migrate our CI integration tests to py39+ version

**Additional information:**

Related to #3928, if we want to run integration tests with pandas v2, we must run on py3.9+,
So that we need to extend our integration tests workflow matrix to run on py310 additionally

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by tokoko** [comment]
Running integration tests only for 3.10 sounds a bit too risky unless we explicitly drop support for lower versions. I think all tests should include the lowest supported python version (3.8 currently, but if we in fact drop it then 3.9)

**Comment by sudohainguyen** [comment]
> Running integration tests only for 3.10 sounds a bit too risky unless we explicitly drop support for lower versions. I think all tests should include the lowest supported python version (3.8 currently, but if we in fact drop it then 3.9)

actually we still do have full integration tests across 3 python versions on master only (after merged), however I agree that we should extend, not replace for now

**Comment by sudohainguyen** [comment]
updated

File: .github/workflows/pr_integration_tests.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.8" ]
+        python-version: [ "3.8", "3.10" ]
```
Comment (franciscojavierarceo) [inline .github/workflows/pr_integration_tests.yml:89]: this purely adds 3.10 support

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -12,8 +12,6 @@ jobs:
         exclude:
           - os: macOS-latest
             python-version: "3.9"
-          - os: macOS-latest
-            python-version: "3.10"
```
Comment (franciscojavierarceo) [inline .github/workflows/unit_tests.yml:16]: did you mean to drop 3.10 here?

Comment (sudohainguyen) [reply .github/workflows/unit_tests.yml:16]: Drop from exclude list, which means we will conduct py310 unit test on macos

**Review by HaoXuAI — APPROVED** [review]
lgtm!

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (franciscojavierarceo) [reply .github/workflows/unit_tests.yml:16]: oh i missed the `exclude` key. Nice.

**Pull request merged**